ldk/javascript/src/search/types.ts

```diff
@@ -0,0 +1,86 @@
+export enum FieldType {
```

Comment:
This enum is called `FieldType`, but the comments refer to a `Filter`. Is that intentional?

Reply:
The `FieldType` is used to describe how we want the field to be interpreted and what we want inserted into the index. The field type mapping themselves apply filters (but do other things as well, like reflection and have specific analyzer for specific data types. The reason I put the filters in the comments is because some of the mappings will filter out certain data (ex english stop words). But we could revisit the naming scheme though, I do see how it can be a bit confusing